core/o11y/src/lib.rs

```diff
@@ -165,36 +170,54 @@ fn is_terminal() -> bool {
     atty::is(atty::Stream::Stderr)
 }
 
-fn make_log_layer<S>(
+fn add_log_layer<S>(
     filter: EnvFilter,
     writer: NonBlocking,
     ansi: bool,
-) -> Filtered<tracing_subscriber::fmt::Layer<S, DefaultFields, Format, NonBlocking>, EnvFilter, S>
+    subscriber: S,
+) -> (
+    Layered<
+        Filtered<
+            fmt::Layer<S, fmt::format::DefaultFields, fmt::format::Format, NonBlocking>,
+            reload::Layer<EnvFilter, S>,
+            S,
+        >,
+        S,
+    >,
```

Comment:
Is there a reason why we couldn’t do the following:

```rust
type LogLayer<Inner> = Layered<Filtered<
    fmt::Layer<Inner, fmt::format::DefaultFields, fmt::format::Format, NonBlocking>,
    reload::Layer<EnvFilter, Inner>,
    Inner,
>, Inner>;
```

And then use `LogLayer<S>` here (and `LogLayer<Registry>` up in the `OTLP_LAYER_RELOAD_HANDLE`)?

Reply:
Didn't know type aliases can be generic.
Thanks!